### ISSUE TITLE
returning "words" option

### ISSUE THREAD

**Issue by pnrobinson** [monarch-initiative/N2V#117]
I am not convinced that returning a 
``tensorflow.python.data.ops.dataset_ops.TensorSliceDataset`` from the Text Encoder class is the best option, because in essence we want just one line of integers, and AFAIK there is no way of getting this directly (cf chapter 15 of the Geron book). It is pretty useful for the list of lists approach that we need for node2vec though!

**Comment by pnrobinson** [comment]
I take it back, I think I figured this out...

**Comment by pnrobinson** [comment]
So I think here it is
```
treasure_island = '/...../treasure-island.txt'

encoder = TextEncoder(treasure_island, data_type='words')
tensor_data,  count_list, dictionary,  reverse_dictionary = encoder.build_dataset()

print("count_list size=%d" % len(count_list))
print("dictionary size=%d" % len(dictionary))
print("reverse_dictionary size=%d" % len(reverse_dictionary))
# print(tensor_data)

print(type(tensor_data))

batch_size = 128
window_shift=1
datawindows = tensor_data.window(size=batch_size)
for wdw in datawindows:
    lst = list(wdw.as_numpy_iterator())
    print(len(lst), lst[0:5])
```

This outputs numpy lists of length 128 until the very last list. We could either skip this or we can process what we have, as long is there is enough for the window span of skip-gram or cbow.
This seems to go decently fast and will not be the bottleneck in code. I think that we can pass this
list to the same batching function that we will use for list-of-lists.

**Comment by callahantiff** [comment]
> So I think here it is
> 
> ```
> treasure_island = '/...../treasure-island.txt'
> 
> encoder = TextEncoder(treasure_island, data_type='words')
> tensor_data,  count_list, dictionary,  reverse_dictionary = encoder.build_dataset()
> 
> print("count_list size=%d" % len(count_list))
> print("dictionary size=%d" % len(dictionary))
> print("reverse_dictionary size=%d" % len(reverse_dictionary))
> # print(tensor_data)
> 
> print(type(tensor_data))
> 
> batch_size = 128
> window_shift=1
> datawindows = tensor_data.window(size=batch_size)
> for wdw in datawindows:
>     lst = list(wdw.as_numpy_iterator())
>     print(len(lst), lst[0:5])
> ```
> 
> This outputs numpy lists of length 128 until the very last list. We could either skip this or we can process what we have, as long is there is enough for the window span of skip-gram or cbow.
> This seems to go decently fast and will not be the bottleneck in code. I think that we can pass this
> list to the same batching function that we will use for list-of-lists.

@pnrobinson - NICE 🎉  that looks great! 

Should we add something to the batch generation function that dynamically calculates the batch size based on the number of input samples?

**Comment by pnrobinson** [comment]
I have timed this and the datawindows solutions takes MUCH longer than just using plain old Python. In the latest revision, I have just used python for the get_batch functions and things seem to be working. I think the main advantage for the datasets API is in managing at a larger scale (e.g., different files).
We should probably implement the get batch functions similar to 
https://www.tensorflow.org/guide/data_performance
with a prefetch option. Let's consider that after we have merged and checked all the branches!
